static/js/activity.js

```diff
@@ -320,23 +321,86 @@ exports.insert_user_into_list = function (user_id) {
     compose_fade.update_one_user_row(elt);
 };
 
+function get_users(search_term) {
+    var all_persons = [];
+    if (search_term.indexOf('') !== -1) {
+        all_persons = presence.get_user_ids();
+    } else {
+        all_persons = people.get_all_persons_ids();
+    }
+    return all_persons;
+}
+
+exports.create_filter_function = function (search_terms) {
+    return function (item) {
+        var person = people.get_person_from_user_id(item);
+
+        // Get person object (and ignore errors)
+        if (!person || !person.full_name) {
+            return;
+        }
+
+        if (people.is_current_user(person.email)) {
+            return false;
+        }
+
+        var match = _.any(search_terms, function (search_term) {
+            return people.person_matches_query(person, search_term);
+        });
+
+        if (match) {
+            return true;
+        }
+        return false;
+    };
+};
+
+exports.retrieve_sidebar_users_ids = function (search_terms) {
+    var users = get_users(search_terms);
+    users = sort_users(users);
+
+    return users;
+};
+
+exports.process_search_terms = function () {
+    var user_list = $(".user-list-filter");
+    var search_term = user_list.expectOne().val().trim();
+    var search_terms = search_term.toLowerCase().split(",");
+    search_terms = _.map(search_terms, function (s) {
+        return s.trim();
+    });
+    return search_terms;
+};
+
+exports.row_modifier = function (item) {
+    var person_data = info_for(item);
+    var row = templates.render('user_presence_row', person_data);
+    return row;
+};
+
 exports.build_user_sidebar = function () {
     if (page_params.realm_presence_disabled) {
         return;
     }
-
-    var user_ids = filter_and_sort(presence.get_user_ids());
-
-    var user_info = _.map(user_ids, info_for);
-    var html = templates.render('user_presence_rows', {users: user_info});
-    $('#user_presences').html(html);
+    var search_terms = exports.process_search_terms();
+    var table = $("#user_presences");
+    var users = exports.retrieve_sidebar_users_ids(search_terms);
+    var filter_function = exports.create_filter_function(search_terms);
+    list_render(table, users, {
+        name: "users_list",
+        modifier: exports.row_modifier,
+        filter: {
+            element: $(".user-list-filter"),
+            callback: filter_function,
+        },
+    }).init();
 
     // Update user fading, if necessary.
     compose_fade.update_faded_users();
 
     resize.resize_page_components();
 
-    return user_info; // for testing
+    return; // for testing
```

Comment:
You can remove the `return` statement here, since it's no longer returning anything for testing.